xmtp_mls/src/groups/mod.rs

```diff
@@ -223,11 +223,11 @@ where
         let to_store = StoredGroup::new(group_id, now_ns(), GroupMembershipState::Pending);
         let stored_group = provider.conn().insert_or_ignore_group(to_store)?;
 
-        Ok(Self::new(
-            client,
-            stored_group.id,
-            stored_group.created_at_ns,
-        ))
+        let xmtp_group = Self::new(client, stored_group.id, stored_group.created_at_ns);
+
+        let _ = xmtp_group.queue_key_update();
```

Comment:
Once this intent is queued, it will cause the key update to be sent the next time `sync()` is run on the group. That will happen even if the member is purely a "listener" to the group.

We really only need to perform this key update for members who are active and sending messages, and so we would like to trigger this action on first message send so that we don't overwhelm the group with unnecessary key updates.

Reply:
I was trying to use the existing `key_update` function which is async, but noticed that the `create_from_welcome` function is sync. That's why I duplicate it to the `queue_key_update` function which is sync (thanks to @richardhuaaa for help) 

Based on your comments, it seems there are still serious concerns and I do think it becomes hard to write test for `queue_key_update` because many other functions are written in async way. I'm wondering what's the best approach to do this without causing other issues? I'm thinking changing all the calling chain to async, but it would be very complicated.

Reply:
It seems I got the point. The key update should be done once the group has the first message sent. Let me think how to do that

Reply:
We really want to do this when the individual installation is about to send their first message to the group.

Reply:
From Daniel and I's discussion, here is how we can do it now:

```
Add column ‘rotated_at_ns’ to groups table
When creating a group from welcome, set ‘rotated_at_ns’ to 0
When sending a message:
  - If ‘rotated_at_ns’ is 0, rotate the key, and then set ‘rotated_at_ns’ to now()
```